applications/antelopeio_firewall.md

```diff
@@ -0,0 +1,232 @@
+
+### Overview
+
+-   **Name:**  Antelope Node Firewall
+-   **Brief Description:**  Lower the barrier to entry for infrastructure providers to offer highly reliable public endpoints.
+-   **Relationship to EOSIO:**  It's a layer on top of existing AntelopeIO API nodes.
+-   **Reason for Interest:**  We want to solve common issues that API node operators have.
+
+### Project Details
+
+#### Problem
+Antelope Leap nodes are easily abused without proper network infrastructure. When nodes are abused by bots and malicious attackers, they become unreliable or unusable for normal users and create considerable strain on infrastructure providers. Therefore today only a small subset of public EOS nodes are usable in production. Based on our custom eospowerup.io proxy that rotates between APIs provided by the top 50 BPs we found that less than 10 are reliable and usable for production applications.
+
+#### Proposed Solution
+We propose an application that would sit behind an existing firewall such as NGINX (or replace your firewall for more simple networks) and in front of a Nodeos API and/or p2p node. The application would monitor incoming transactions and queries to perform smart logic such as blocking, or prioritizing based on the node operator's preference. The purpose is to make running highly available RPC nodes more accessible to node operators and reduce the load on the Nodeos software. The advantage of this application vs a traditional firewall is that this application is aware of Antelope transaction and query structure and can apply complex filter logic that a traditional firewall isn’t designed for.  
+
+![AntelopeIO_Firewall_Diagram](https://ipfs.animus.is/ipfs/QmZ3qEwAGuaboNYAypJdXtcGnofQX4yPrYhVStQnX5u8Vh?filename=diagram-antelope-firewall.jpg)
+
+The initial version of the application would be written in Rust and could be easily scaled across multiple cores and operated with high uptime. The application could run on the same VM as Leap but for optimal flexibility could be hosted on a dedicated VM that scales independently.
+
+#### **Features**
+
+- **Dynamic Blacklisting**  Accounts, IPs, keys, and even specific contracts and actions can be blacklisted, throttled, and subjectively billed based on granular configuration. For example: the node operator may opt to only process one transfer action per account per second. For accounts that use inline actions or advanced techniques to subvert the firewall filtering, subjective CPU billing based on recent transaction history can effectively prevent abuse. All this logic can be easily configured by the node operator and dynamically automated based on metrics such as node CPU/RAM threshold logic. A side effect of this is that application specific nodes could provide fast and reliable operation for a limited subset of actions related to their application while throttling or deprioritizing other traffic.
+
```

Comment:
Milestone 1 deliverable should contain detailed description of the static blacklist configuration formats and dynamic blacklisting algorithms.

Reply:
Based on feedback from alpha/beta users during milestone 2 development the details will become solidified. Our documentation will cover all the details.